server.go

```diff
@@ -2069,9 +2069,20 @@ func (s *Server) createShardGroupsIfNotExists(database, retentionPolicy string,
 		// Local function makes locking fool-proof.
 		s.mu.RLock()
 		defer s.mu.RUnlock()
+
+		rp, err := s.RetentionPolicy(database, retentionPolicy)
+		if err != nil {
+			return err
+		}
+
+		times := map[time.Time]struct{}{}
 		for _, p := range points {
+			times[p.Time.Truncate(rp.ShardGroupDuration)] = struct{}{}
```

Comment:
i have a question to try and improve my own understanding of go.  i've seen this pattern a couple of times in this codebase so i have to ask...

since you only care about the keys of this map, is it any more efficient (from the point of view of memory allocation) to make this a `map[time.Time]* struct{}{}` and use `nil` as the value?